File: 字符串/子序列/392.判断子序列/392.py

def isSubsequence(s, t):
    s_length = len(s)
    t_length = len(t)
    i = j = 0

    while i < s_length and j < t_length:
        if s[i] == s[j]:
            i += 1
        j += 1
    
    return i == s_length

File: 字符串/子序列/392.判断子序列/392.c

bool isSubsequence(char* s, char* t) {
    int s_length = strlen(s);
    int t_length = strlen(t);
    int i = 0, j = 0;
    while (i < s_length && j < t_length) {
        if (s[i] == t[j]) {
            i++;
        }
        j++;
    }

    return i == s_length;
}